source/arm7/firmware.c

```diff
@@ -39,6 +39,21 @@ void readFirmware(u32 address, void *destination, u32 size)
     leaveCriticalSection(oldIME);
 }
 
+void readJEDEC(u8 *destination, u32 size) {
```

Comment:
This function should return an ``int``: 0 on success, -1 (for example) on error. The other functions in this file should do that too, but let's ignore that for the time being.

Also, you should check if ``destination`` is NULL.

Reply:
I kind of arbitrarily added error codes for destination and size. (Call me incompetent, but never actually implemented error codes in an expected way before.)